pattern.c

```diff
@@ -1361,15 +1361,18 @@ void mutt_pattern_free(struct Pattern **pat)
     mutt_pattern_free(&tmp->child);
     FREE(&tmp);
   }
+
+
+  FREE(pat);
 }
 
-/**
- * mutt_pattern_new - Create a new Pattern
- * @retval ptr New Pattern
- */
-struct Pattern *mutt_pattern_new(void)
+static struct PatternHead *mutt_pattern_new_node(void)
```

Comment:
Please can you add a Doxygen comment block for this function.

Reply:
Done.